pkg/config/control_plane.go

```diff
@@ -167,6 +173,21 @@ func (d *ControlPlaneDataStore) UnmarshalJSON(data []byte) error {
 		if len(gc.Config) > 0 {
 			err = json.Unmarshal(gc.Config, d.MySQLConfig)
 		}
+	case model.DataStoreGCS:
+		d.GCSConfig = &FileStoreGCSConfig{}
+		if len(gc.Config) > 0 {
+			err = json.Unmarshal(gc.Config, d.GCSConfig)
+		}
+	case model.DataStoreS3:
+		d.S3Config = &FileStoreS3Config{}
+		if len(gc.Config) > 0 {
+			err = json.Unmarshal(gc.Config, d.S3Config)
+		}
+	case model.DataStoreMINIO:
+		d.MinioConfig = &FileStoreMinioConfig{}
+		if len(gc.Config) > 0 {
+			err = json.Unmarshal(gc.Config, d.MinioConfig)
+		}
```

Comment:
As we talked in the Slack, let's simplify this part.

Reply:
Thanks for your comment 🙌  Addressed by https://github.com/pipe-cd/pipecd/pull/3455/commits/f8c525a0831f9cc859e4c69ba4e2c5dec6f0f60f 🙏